subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -1080,19 +1219,129 @@ static uint32_t filter_remove(struct lll_filter *filter, uint8_t addr_type,
 }
 #endif /* !CONFIG_BT_CTLR_PRIVACY */
 
-static void filter_insert(struct lll_filter *filter, int index, uint8_t addr_type,
-			   uint8_t *bdaddr)
+static void filter_insert(struct lll_filter *filter, int index,
+			  uint8_t addr_type, uint8_t *bdaddr)
 {
 	filter->enable_bitmask |= BIT(index);
 	filter->addr_type_bitmask |= ((addr_type & 0x01) << index);
-	memcpy(&filter->bdaddr[index][0], bdaddr, BDADDR_SIZE);
+	(void)memcpy(&filter->bdaddr[index][0], bdaddr, BDADDR_SIZE);
 }
 
 static void filter_clear(struct lll_filter *filter)
 {
 	filter->enable_bitmask = 0;
 	filter->addr_type_bitmask = 0;
 }
+#endif /* CONFIG_BT_CTLR_FILTER */
+
+#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST)
+static void pl_clear(void)
+{
+	for (int i = 0; i < PL_SIZE; i++) {
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		uint8_t j = pl[i].rl_idx;
+
+		if (j < ARRAY_SIZE(pl)) {
+			rl[j].pl = 0U;
+		}
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+
+		pl[i].taken = 0U;
+	}
+}
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+static uint8_t pl_addr_find(uint8_t addr_type, uint8_t *addr)
+{
+	for (int i = 0; i < PL_SIZE; i++) {
+		if (PL_ADDR_MATCH(addr_type, addr)) {
+			return i;
+		}
+	}
+
+	return FILTER_IDX_NONE;
+}
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+
+static uint8_t pl_find(uint8_t addr_type, const uint8_t *addr, uint8_t sid,
+		       uint8_t *free_idx)
```

Comment:
In `ull_filter_rl_find` the `free_idx` argument's type is `uint8_t *const free_idx`. Can/should that be applied here as well?

Reply:
done